MatrixGenerators/BlockMatrix.py

```diff
@@ -128,6 +139,14 @@ def _generate_symmetric_gaussian_block_matrix(mean: float,
         max_block_index = (cluster_index + 1) * cluster_size
         result_matrix[min_block_index: max_block_index, min_block_index: max_block_index] = single_matrix_block
 
+    if remainder > 0:
```

Comment:
add TODO to fix block sizes calculation and generation. need to pass agents_num and calc sizes here. no need to pass remainder.

fix code replication of for and if.